web/modules/custom/weather/src/Services/WeatherApi.php

```diff
@@ -2,31 +2,46 @@
 
 namespace Drupal\weather\Services;
 
+use Drupal\Component\DependencyInjection\ContainerInterface;
 use Drupal\Core\Config\ConfigFactoryInterface;
 use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
 
 /**
  * Service class WeatherApi.
  */
 class WeatherApi {
 
   /**
-   * Constructs a ConfigFactory and a Client object.
+   * Constructs a new WeatherApi object.
    */
   public function __construct(
     protected ConfigFactoryInterface $configFactory,
     protected ClientInterface $httpClient,
-  ) {}
+  ) {
+  }
+
+  /**
+   * Creates a new instance of WeatherApi.
+   */
+  public static function create(ContainerInterface $container): static {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('http_client')
+    );
+  }
```

Comment:
In services, you don't need the create() method. In the services.yml file, you pass these services as arguments and essentially replace this method